File: robocode-tankroyale-bot-api-java/src/main/java/dev/robocode/tankroyale/botapi/events/GameStartedEvent.java
package dev.robocode.tankroyale.botapi.events;

import dev.robocode.tankroyale.botapi.GameSetup;

/** Event occurring when game has just started */
@SuppressWarnings("unused")
public final class GameStartedEvent implements IMessage {

  /** The ID used for identifying your bot in the current battle */
  private final int myId;

  /** The game setup for the battle just started */
  private final GameSetup gameSetup;

  public GameStartedEvent(int myId, GameSetup gameSetup) {
    this.myId = myId;
    this.gameSetup = gameSetup;
  }

  /** Returns the ID used for identifying your bot in the current battle */
  public int getMyId() {
    return myId;
  }

  /** Returns the game setup for the battle just started */
  private GameSetup getGameSetup() {
    return gameSetup;
  }
}

File: robocode-tankroyale-bot-api-csharp/src/events/GameStartedEvent.cs
using Newtonsoft.Json;

namespace Robocode.TankRoyale.BotApi
{
  /// <summary>
  /// Event occurring when game has just started.
  /// </summary>
  public sealed class GameStartedEvent : IMessage
  {
    /// <summary>The ID used for identifying your bot in the current battle.</summary>
    public int MyId { get; }

    /// <summary>The game setup for the battle just started.</summary>
    public GameSetup GameSetup { get; }

    /// <summary>
    /// Initializes a new instance of the GameStartedEvent class.
    /// </summary>
    /// <param name="myId">ID used for identifying your bot in the current battle.</param>
    /// <param name="gameSetup">Game setup for the battle just started.</param>
    [JsonConstructor]
    public GameStartedEvent(int myId, GameSetup gameSetup) : base() =>
      (MyId, GameSetup) = (myId, gameSetup);
  }
}